### ISSUE TITLE
Automatic query queueing?

### ISSUE THREAD

**Issue by chichilatte** [GetBreeze/breeze-db#1]
Hi, love this library, thank you so much! 

I'm wondering how it deals with queueing of queries. E.g. I run a `BreezeRawQuery.multiQueryTransaction()` which takes 10 seconds, but after 5 seconds I run another query. I'm imagining I'd get a `DB locked` error, but maybe the queries are automatically queued? That wld be awesome.

Sorry, I should just run a few tests. Just hoped you might have a quick answer!

**Comment by ecornbower** [comment]
Thanks for the kind feedback!

In your example the second query should be automatically queued by SQLite until the first one is finished. @marpies can you confirm this behavior?

**Comment by marpies** [comment]
There is no queueing mechanism implemented in BreezeDb and you will need to wait for the callback to make sure the queries are executed in order. Right now you may run into problems if you try to run queries the way you described, so it certainly would be very useful to have BreezeDb control the queue automatically.

Thanks for bringing this up, we'll consider adding this feature.

**Comment by ecornbower** [comment]
@marpies Thanks for the clarification. Will BreezeDB throw an error in this case, or is it just an issue of query order not necessarily being honored?

**Comment by marpies** [comment]
@kcornbower Yes, the order is not being honored and that can lead to unexpected results if the developer expects the queries to be queued / ran in the order the calls are made. There is no error thrown.

**Comment by ecornbower** [comment]
Thanks @marpies that's what I figured. I'm fine if you want to put this feature on the back-burner and come back to it in the future when you have some free time.

@chichilatte Does that answer everything for you?

**Comment by chichilatte** [comment]
Thank you both, that's great to know. I can't find any official as3 docs on query queuing, you'd think it wld be quite important. I implemented a really basic yet clunky queueing system for our project, not too hard. If I have anything useful to contribute back to the project I will (already forked it to add `compact()` and `loadSchema()` to `BreezeSchemaBuilder`).

FYI Right now we're developing an app that's using `com.probertson.data.SQLRunner` with some queuing. It's giving us occasional `SQL Error: Error #3119: Database file is currently locked.` errors when the db gets really hammered (as well as a listener memory leak). So it may not just be ordering that gets screwed. I've replaced that library with Breeze+proper queueing and no more errors so far :)
